base/some.jl

```diff
@@ -62,13 +76,20 @@ julia> coalesce(nothing, Some(1))
 """
 function coalesce end
 
-coalesce(x::Any) = x
-coalesce(x::Some) = x.value
-coalesce(x::Nothing) = nothing
-coalesce(x::Missing) = missing
-coalesce(x::Any, y...) = x
-coalesce(x::Some, y...) = x.value
-coalesce(x::Union{Nothing, Missing}, y...) = coalesce(y...)
+coalesce(x::Nothing, y) = y
+coalesce(x::Missing, y) = y
+coalesce(x::Nothing, y::Some) = y.value
+coalesce(x::Missing, y::Some) = y.value
+for T in (:Nothing, :Missing)
+    @eval begin
+        coalesce(x::$T, y::Union{Nothing, Missing}) =
+            throw(ArgumentError("coalesce requires that least one argument differs from `nothing` or `missing`; " *
```

Comment:
The line here is pretty long. I'd suggest changing the semicolon to a period and the space after it to a newline. Not a big deal though.

Reply:
Fast! I wasn't sure what's the convention about this. It's indeed problematic in the doctest.

Reply:
Fixed.